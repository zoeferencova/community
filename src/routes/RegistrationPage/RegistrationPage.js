import React, { Component } from "react";
import { Link } from "react-router-dom";
import RegistrationForm from "../../components/RegistrationForm/RegistrationForm";
import AuthApiService from "../../services/auth-api-service";
import styles from "../LoginPage/LoginPage.module.css";

export default class RegistrationPage extends Component {
    state = {
        loading: false,
        error: null
    }

    register = (first_name, email, password, confirm_password) => {
        this.setState({...this.state, loading: true });
        password !== confirm_password
            ? this.setState({ error: "Passwords do not match", loading: false })
            : AuthApiService.postUser({
                first_name: first_name,
                email: email,
                password: password
            })
                .then(user => {
                    this.setState({...this.state, loading: false });
                    this.props.history.push({
                        pathname: "/login",
                        success: true
                    });
                })
                .catch(res => 
                    this.setState({ error: res.error, loading: false })
                )

    }
    
    render() {
        return (
            <div className={styles.container}>
                <div className={styles.main}>
                    <h1>Sign up</h1>
                    <div className={styles.form}>
                        <RegistrationForm loading={this.state.loading} error={this.state.error} register={this.register} updateSuccess={this.props.updateSuccess} />
                        <p>Already have an account? <Link to="/login">Sign in</Link></p>
                    </div>
                </div>     
            </div>
        )
    }
}