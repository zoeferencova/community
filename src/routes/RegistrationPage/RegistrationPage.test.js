import React from "react";
import { render } from "enzyme";
import RegistrationPage from "./RegistrationPage";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("RegistrationPage", () => {
    it("renders RegistrationPage component given mock context data", () => {
        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <BrowserRouter>
                    <RegistrationPage />
                </BrowserRouter>
            </CommUnityContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});