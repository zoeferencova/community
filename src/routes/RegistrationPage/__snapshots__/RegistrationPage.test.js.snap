// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage renders RegistrationPage component given mock context data 1`] = `
<div
  class="container"
>
  <div
    class="main"
  >
    <h1>
      Sign up
    </h1>
    <div
      class="form"
    >
      <form>
        <label
          class="label "
          for="first_name"
        >
          First name
        </label>
        <input
          class="input "
          id="first_name"
          name="first_name"
          required=""
          type="text"
        />
        <label
          class="label "
          for="email"
        >
          Email
        </label>
        <input
          class="input "
          id="email"
          name="email"
          required=""
          type="email"
        />
        <label
          class="label "
          for="password"
        >
          Password
        </label>
        <input
          class="input "
          id="password"
          name="password"
          required=""
          type="password"
        />
        <label
          class="label "
          for="confirm_password"
        >
          Confirm Password
        </label>
        <input
          class="input "
          id="confirm_password"
          name="confirm_password"
          required=""
          type="password"
        />
        <div
          class="buttonContainer"
        >
          <button
            class="button "
            type="button"
          >
            Cancel
          </button>
          <button
            class="buttonDark registerButton"
            loading="false"
            type="submit"
          >
            Create Account
          </button>
        </div>
      </form>
      <p>
        Already have an account? 
        <a
          href="/login"
        >
          Sign in
        </a>
      </p>
    </div>
  </div>
</div>
`;
